很赞的功能,但是在luajit2.1下无法使用

现在我们后端用的是openresty,打算把这个功能加入到openresty中.
尝试编译了下,发现不支持luajit2.1啊.....
一些5.2下的语法jit还是不支持....
这块xjdrew大大能提供支持么....

luaL_checkversion(L);

lua-skiplist.c: In function ‘_delete’:
lua-skiplist.c:40: warning: assignment discards qualifiers from pointer target type
lua-skiplist.c: In function ‘_delete_by_rank’:
lua-skiplist.c:56: warning: implicit declaration of function ‘luaL_checkunsigned’
lua-skiplist.c:65: warning: implicit declaration of function ‘lua_pushunsigned’
lua-skiplist.c: In function ‘_get_rank’:
lua-skiplist.c:82: warning: assignment discards qualifiers from pointer target type
lua-skiplist.c: In function ‘luaopen_skiplist_c’:
lua-skiplist.c:182: warning: implicit declaration of function ‘luaL_checkversion’
lua-skiplist.c:200: warning: implicit declaration of function ‘luaL_newlib’

--- a/Makefile
+++ b/Makefile
@@ -5,6 +5,9 @@ CFLAGS = -g3 -O0 -Wall -fPIC --shared
 LUA_INCLUDE_DIR = /usr/local/include
 DEFS = -DLUA_COMPAT_5_2
 
+luajit: LUA_INCLUDE_DIR = /usr/local/include/luajit-2.1
+luajit: skiplist.so
+
 skiplist.so: skiplist.h skiplist.c lua-skiplist.c
 	$(CC)  $(CFLAGS)  -I$(LUA_INCLUDE_DIR) $(DEFS)  $^ -o $@
 

--- a/lua-skiplist.c
+++ b/lua-skiplist.c
@@ -191,7 +191,9 @@ _release(lua_State *L) {
 }
 
 int luaopen_skiplist_c(lua_State *L) {
+#if defined(LUA_VERSION_NUM) && LUA_VERSION_NUM > 501
     luaL_checkversion(L);
+#endif
 
     luaL_Reg l[] = {
         {"insert", _insert},

